src/cpp/src/generation_handle.cpp

```diff
@@ -46,8 +46,11 @@ std::vector<GenerationOutput> GenerationHandleImpl::read_all() {
         add_partial_result(partial_results, iteration_results);
     }
 
-    for (auto& partial_result: partial_results) {
+    for (auto& partial_result : partial_results) {
         results.push_back(partial_result.second);
+        if (results.size() == m_sampling_params.num_return_sequences) {
+            break;
+        }
```

Comment:
Results returned by read() and collected in read_all() are unordered. For returning `n` sequences I guess we want those with the highest score right?

Reply:
Beams should be sorted within same prompt group
https://github.com/openvinotoolkit/openvino.genai/blob/97a05e1eb733662d639c2fd57bab01c11fa3da27/src/cpp/src/group_beam_searcher.cpp#L462

Reply:
@Wovchena you point to code which is not related to CB

Reply:
That's the point. Beams order should be sorted in CB just like in non CB